src/org/gk/gkCurator/CuratorActionCollection.java

```diff
@@ -2169,10 +2169,41 @@ private void deleteInstance() {
 		}
 	}
 	
+	/**
+	 * Retrieve a "Create Instance" action for new entities.
+	 * 
+	 * @return Action with specific "SHORT_DESCRIPTION" and "enabled" status
+	 */
 	public Action getCreateInstanceAction() {
-		if (createInstanceAction == null) {
+		// list of constants that curators are not allowed to create new instances of.
+		ArrayList<String> nonCreatable = new ArrayList<String>();
+		nonCreatable.add(ReactomeJavaConstants.Species);
+		nonCreatable.add(ReactomeJavaConstants.Compartment);
+
+		// get selected class.
+		GKSchemaClass selectedClass = null;
+		if (curatorFrame.getFocusedComponent() instanceof SchemaViewPane) {
+			SchemaViewPane schemaViewPane = (SchemaViewPane) curatorFrame.getFocusedComponent();
+			selectedClass = schemaViewPane.getSchemaPane().getSelectedClass();
+		}
+
+		// if creation of instance is not allowed, then user is notified to contact editor.
+		if (selectedClass != null && nonCreatable.contains(selectedClass.getName())) {
+			// This is an information for blocking creating a new species and compartment:
+			String message = "You cannot create an instance for this type of class. Please ask Peter D’Eustachio to create one for you.";
```

Comment:
Maybe the person to contact could be made a constant in this class or another class managing constant values?

Reply:
I will migrate the name over to a constant value class!

Do you think the uneditable classes (`species` and `compartment` in this example), be in a separate class as well to keep things easier to maintain?

Reply:
Liam, As we discussed, these values (un-editable classes) should be configured inside the file resources/curator.xml. If you like, you can also configure the contact name.